addOns/formhandler/src/test/java/org/zaproxy/zap/extension/formhandler/FormHandlerValueGeneratorUnitTest.java

```diff
@@ -61,4 +64,17 @@ void shouldUseDefaultValueWhenAvailableForUnidentifiedFields(String fieldId) {
         // Then
         assertThat(generatedvalue, is(equalTo(defaultValue)));
     }
+
+    @Test
+    void shouldUseRegexWhenAppropriate() {
+        // Given
+        String value = "foobar";
+        given(param.getConfig()).willReturn(new ZapXmlConfiguration());
+        given(param.getEnabledFieldValue("foo")).willReturn("alice");
+        given(param.getEnabledFieldValue("bar")).willReturn(value);
```

Comment:
Is there a less fake way for me to test this? I mean really this is just testing that I properly told the test how to behave :wink: